operate/schema/src/main/java/io/camunda/operate/store/elasticsearch/NewElasticsearchBatchRequest.java

```diff
@@ -0,0 +1,306 @@
+            .index(index)
+            .id(id)
+            .action(a -> a.doc(updateFields))
+            .retryOnConflict(UPDATE_RETRY_COUNT)
+            .build();
+    bulkRequestBuilder.operations(op -> op.update(updateOperation));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest update(final String index, final String id, final OperateEntity entity)
+      throws PersistenceException {
+    LOGGER.debug("Add update request for index {} id {} and entity {}", index, id, entity);
+
+    final var updateOperation =
+        new UpdateOperation.Builder<OperateEntity, OperateEntity>()
+            .index(index)
+            .id(id)
+            .action(a -> a.doc(entity))
+            .retryOnConflict(UPDATE_RETRY_COUNT)
+            .build();
+    bulkRequestBuilder.operations(op -> op.update(updateOperation));
+
+    return this;
+  }
+
+  @Override
+  public BatchRequest updateWithScript(
+      final String index,
+      final String id,
+      final String script,
+      final Map<String, Object> parameters)
+      throws PersistenceException {
+    LOGGER.debug(
+        "Add upsert request with for index {} id {} and script {} with parameters {} ",
+        index,
+        id,
+        script,
+        parameters);
+
+    final var updateOperation =
+        new UpdateOperation.Builder<OperateEntity, Map<String, Object>>()
+            .index(index)
+            .id(id)
+            .action(a -> a.script(getScriptWithParameters(script, parameters)))
+            .retryOnConflict(UPDATE_RETRY_COUNT)
+            .build();
+    bulkRequestBuilder.operations(op -> op.update(updateOperation));
+
+    return this;
+  }
+
+  @Override
+  public void execute() throws PersistenceException {
+    execute(false);
+  }
+
+  @Override
+  public void executeWithRefresh() throws PersistenceException {
+    execute(true);
+  }
+
+  private void execute(final boolean shouldRefresh) throws PersistenceException {
+    if (shouldRefresh) {
+      bulkRequestBuilder.refresh(Refresh.True);
+    }
+    final BulkRequest bulkRequest = bulkRequestBuilder.build();
+    try {
+      LOGGER.debug("************* FLUSH BULK START *************");
+      final BulkResponse bulkResponse = esClient.bulk(bulkRequest);
+      final List<BulkResponseItem> items = bulkResponse.items();
+      for (final BulkResponseItem item : items) {
+        if (item.error() != null) {
+          LOGGER.error(
+              String.format(
+                  "Bulk request execution failed. %s. Cause: %s.", item, item.error().reason()));
+          throw new PersistenceException("Operation failed: " + item.error().reason());
+        }
+      }
+      LOGGER.debug("************* FLUSH BULK FINISH *************");
+    } catch (final IOException ex) {
+      throw new PersistenceException(
+          "Error when processing bulk request against Elasticsearch: " + ex.getMessage(), ex);
+    }
+  }
+
+  private static Script getScriptWithParameters(
```

Comment:
I think we need to consider a way to split the scriptbuilding out into a separate helper class if at all possible to avoid growing this class too much with a bunch of helper functions. This function and the associated json functions could be in a script transformation helper class, which we could utilize in one of two ways:

1. Pass an instance of this new helper class into the constructor via dependency injection, and call it here inside getScriptWithParameters

2. Create a new interface method with an alternate signature for updateWithScript, where we pass in the already-finalized script instead of the script template with parameters. This helper class would be called outside the batch request in the ExportHandler before it calls upsertWithScript. It would require an empty implementation in the legacy classes but my understanding is we are eventually removing those anyway. This would be my preferred approach.

Thoughts?

Reply:
I am not familiar with your definition of helper class. Do you mean a class with public static methods ? Or non-static, where we have to create an instance every time we want to use it ? Do you have examples maybe ? Thank you.

Reply:
I updated.